### DISCUSSION TITLE
Sanitization process when loading from sd files doesn't preserve hydrogens

### DISCUSSION THREAD

**Discussion by zarkoivkovicc** [rdkit/rdkit#7102] [Q&A]
When loading molecules from nmrshiftdb2 sd file, even though I specify sanitize=True and removeHs=False, the molecules have radicals instead of hydrogens. Atoms have no implicit or explicit hydrogens and it has explicit valences equal to the actual values - number of unpaired electrons for that atom. Function AddHs doesn't do anything. Is there something obvious I am missing here? When I set sanitize=False, it doesn't produce radicals, but it's impossible to run AddHs() on molecules that are not sanitized.

**Comment by greglandrum** [comment]
Can you please provide an example file along with the exact commands you are using to load it?

**Comment by zarkoivkovicc** [comment]
You can download the sd file here: https://sourceforge.net/projects/nmrshiftdb2/files/data/nmrshiftdb2withsignals.sd/download
```from rdkit import Chem
suppl = Chem.SDMolSupplier('nmrshiftdb2withsignals.sd',sanitize=True, removeHs=False, strictParsing=True)
data = list(suppl)
for mol in data[:10]:
    if mol:
        display(mol)
```
